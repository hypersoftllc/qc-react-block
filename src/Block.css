.Block {
  box-sizing: border-box;
  display: block;
  overflow: auto;
}
