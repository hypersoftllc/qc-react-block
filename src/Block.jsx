import PT from 'prop-types'
import React from 'react'


function Block(props) {
  const domProps = {
    ...props,
    className: `${props.compClassName} ${props.className}`,
  }
  delete domProps.children
  delete domProps.compClassName
  delete domProps.compType
  return React.createElement(
    props.compType,
    domProps,
    props.children,
  )
}

Block.defaultProps = {
  className: '',
  compClassName: 'Block',
  compType: 'div',
}

Block.propTypes = {
  children: PT.node,
  className: PT.string,
  compClassName: PT.string,
  compType: PT.oneOf([
    'address',
    'article',
    'aside',
    'blockquote',
    'details',
    'div',
    'dl',
    'fieldset',
    'figcaption',
    'figure',
    'footer',
    'form',
    'header',
    'main',
    'nav',
    'section',
  ]),
}

export default Block
