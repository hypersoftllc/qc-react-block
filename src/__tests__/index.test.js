import React from 'react'
import TestRenderer from 'react-test-renderer'

import Block from '../'


describe('Block', () => {

  it('should be a function', () => {
    expect(typeof Block).toBe('function')
  })

  it('should be a functional component', () => {
    let comp

    comp = TestRenderer.create(
      <Block/>
    )

    // Functional components don't have instances.
    expect(comp.getInstance()).toBe(null)
  })

  describe('rendered', () => {

    describe('with no props', () => {
      
      it('should use default props', () => {
        let comp, tree

        comp = TestRenderer.create(
          <Block/>
        )

        tree = comp.toJSON()
        expect(tree).toMatchSnapshot()
      })

    })

    describe('with children', () => {
      
      it('should render children', () => {
        let comp, tree

        comp = TestRenderer.create(
          <Block>
            <span>Child</span>
          </Block>
        )

        tree = comp.toJSON()
        expect(tree).toMatchSnapshot()
      })

    })

    describe('with className', () => {
      
      it('should render with expected class attribute', () => {
        let comp, tree

        comp = TestRenderer.create(
          <Block className="Foo"/>
        )

        tree = comp.toJSON()
        expect(tree).toMatchSnapshot()
      })

    })

    describe('with compClassName', () => {
      
      it('should render with expected class attribute', () => {
        let comp, tree

        comp = TestRenderer.create(
          <Block compClassName=""/>
        )

        tree = comp.toJSON()
        expect(tree).toMatchSnapshot()

        comp = TestRenderer.create(
          <Block className="foo" compClassName=""/>
        )

        tree = comp.toJSON()
        expect(tree).toMatchSnapshot()

        comp = TestRenderer.create(
          <Block className="foo" compClassName="block"/>
        )

        tree = comp.toJSON()
        expect(tree).toMatchSnapshot()
      })

    })
  
  })

})
