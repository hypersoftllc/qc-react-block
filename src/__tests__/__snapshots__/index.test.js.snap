// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Block rendered with children should render children 1`] = `
<div
  className="Block"
>
  <span>
    Child
  </span>
</div>
`;

exports[`Block rendered with className should render with expected class attribute 1`] = `
<div
  className="Block Foo"
/>
`;

exports[`Block rendered with compClassName should render with expected class attribute 1`] = `
<div
  className=""
/>
`;

exports[`Block rendered with compClassName should render with expected class attribute 2`] = `
<div
  className="foo"
/>
`;

exports[`Block rendered with compClassName should render with expected class attribute 3`] = `
<div
  className="block foo"
/>
`;

exports[`Block rendered with no props should use default props 1`] = `
<div
  className="Block"
/>
`;
